src/plugin_discovery/plugin_pack/package_json_at_path.js

```diff
@@ -0,0 +1,37 @@
+import { readFileSync } from 'fs';
+import { Observable } from 'rxjs';
+import { resolve } from 'path';
+import { createInvalidPackError } from '../errors';
+
+import { isDirectory } from './lib';
+
+async function createPackageJsonAtPath(path) {
+  if (!await isDirectory(path)) {
+    throw createInvalidPackError(path, 'must be a directory');
+  }
+
+  let str;
+  try {
+    str = readFileSync(resolve(path, 'package.json'));
+  } catch (err) {
+    throw createInvalidPackError(path, 'must have a package.json file');
+  }
+
+  let pkg;
+  try {
+    pkg = JSON.parse(str);
+  } catch (err) {
+    throw createInvalidPackError(path, 'must have a valid package.json file');
+  }
+
+  return {
+    directoryPath: path,
+    contents: pkg,
+  };
+}
+
+export const createPackageJsonAtPath$ = (path) => (
+  Observable.defer(() => createPackageJsonAtPath(path))
```

Comment:
Yeah... there's a trend of referring to a function that creates an observable as `create${something}$` with the postfix $ denoting that it's an Observable (borrowed from the RxJs community). I'm not partial to the name, perhaps `scanPackageJsonAtPath$`?

Reply:
Probably not enough to justify the naming scheme, since I think it will continue to trip up people who are exposed to it for the first time, but the idea is that when you read `$` at the end of a name it stands for "stream", so the method name is intended to be read as "create packageJsonAtPath stream", which I think helps avoid the assumption that this is creating package.json files on the file system.

Reply:
Interesting, the naming makes a lot more sense when reading the "$" as "stream"